supervisor/jira_utils.py

```diff
@@ -367,26 +405,55 @@ def get_current_issues(
     jql: str,
     full: bool = False,
 ) -> Generator[Issue, None, None] | Generator[FullIssue, None, None]:
-    start_at = 0
     max_results = 1000
-    while True:
-        body = {
-            "jql": jql,
-            "startAt": start_at,
-            "maxResults": max_results,
-            "fields": _fields(full),
-        }
-
-        logger.debug("Fetching JIRA issues, start=%d, max=%d", start_at, max_results)
-        response_data = jira_api_post("search", json=body, decode_response=True)
-        logger.debug("Got %d issues", len(response_data["issues"]))
-
-        for issue_data in response_data["issues"]:
-            yield decode_issue(issue_data, full)
 
-        start_at += max_results
-        if response_data["total"] <= start_at:
-            break
+    if is_jira_cloud():
+        # Cloud: Use v3 search/jql endpoint (v2 is deprecated)
+        next_page_token = None
+        while True:
+            body = {
+                "jql": jql,
+                "maxResults": max_results,
+                "fields": _fields(False),
```

Comment:
Should probably be `[] if full else _fields(False)` (with an appropriate comment)